desktop-widgets/divelogexportdialog.cpp

```diff
@@ -414,23 +367,29 @@ void DiveLogExportDialog::export_TeX(const char *filename, const bool selected_o
 		put_format(&buf, "\\def\\divemaster{%s}\n", dive->divemaster ? dive->divemaster : "");
 		put_format(&buf, "\\def\\suit{%s}\n", dive->suit ? dive->suit : "");
 
-		//Code block prints start/end press. for all cylinders defined in dive, number of cyl. used, SAC, and total delta_p
+		// Print cylinder data
 		put_format(&buf, "\n%% Gas use information:\n");
 		qty_cyl = 0;
 		for (i = 0; i < MAX_CYLINDERS; i++){
 
-			/* Previously, this IF statement used "is_cylinder_used(...)" to print only the cylinders where pressure
-			 * identified by Subsurface as being used.  Now it prints all cylinders that are associated with the
-			 * dive. This way, bailout cylinders that are not needed are still shown, rather than disappearing,*/
-			if (dive->cylinder[i].type.description != NULL){
+			if (is_cylinder_used(dive, i) || (prefs.display_unused_tanks && dive->cylinder[i].type.description)){
 				put_format(&buf, "\\def\\cyl%ddescription{%s}\n", i + 1, dive->cylinder[i].type.description);
-				put_format(&buf, "\\def\\cyl%dmixO2{%.1f\\%}\n", i + 1, get_o2(&dive->cylinder[i].gasmix)/10.0);
-				put_format(&buf, "\\def\\cyl%dmixHe{%.1f\\%}\n", i + 1, get_he(&dive->cylinder[i].gasmix)/10.0);
-				put_format(&buf, "\\def\\cyl%dmixN2{%.1f\\%}\n", i + 1, (100.0 - (get_o2(&dive->cylinder[i].gasmix)/10.0) - (get_he(&dive->cylinder[i].gasmix)/10.0)));
+				put_format(&buf, "\\def\\cyl%dmixO2{%.1f\\%%}\n", i + 1, get_o2(&dive->cylinder[i].gasmix)/10.0);
```

Comment:
Please see below for a suggestion on how to circumvent this problem by using letters instead of numbers.